### ISSUE TITLE
Unsupported Devices in HA (Kogan Smart Blinds and Brilliant Smart Sensor Light)

### ISSUE THREAD

**Issue by comangus13** [tuya/tuya-home-assistant#754]
Please hepl with below two devices. I was able to cofigured in tuya API, but its showing as Unsopported in Home Assistante
HA Version: core-2021.11.5

****Device normal info**
- Brand - Kogan Smart Blinds**
Device Information
Product Name: Smart Blinds
Device ID: 05305504a4cf12d41ea0
Product Category: cl
Device Status: Online
Activated At: 2021-08-09 10:13:07

Device Association
Project: Home Assistance
Association: comangus13@hotmail.com
Data Center: Central Europe Data Center

Extension Information
Sub-Device or Not: No
IP Address: 124.177.209.44

****Device normal info**
- Brand - **Brilliant Smart Sensor Light**

Device Information
Product Name: Brilliant Smart Sensor Light
Device ID: bf10ea3ba7c8f0c8e0knnj
Product Category: qt
Device Status: Online
Activated At: 2020-02-08 22:41:57

Device Association
Project: Home Assistance
Association: comangus13@hotmail.com
Data Center: Central Europe Data Center

Extension Information
Sub-Device or Not: No
IP Address: 124.177.209.44

**Comment by frenck** [comment]
Yesterday, Home Assistant 2022.2 was released.
Could you try upgrading to that version and see if the issue has been resolved?

If not, Home Assistant 2022.2 contains a new diagnostic tool. There is now a "Download Diagnostics" button on each Tuya device on the device page. Clicking that button will download diagnostics information for that specific device. Drag the downloaded file into this issue. It will help with finding out what the problem is.

**Comment by rackxp** [comment]
Hi @frenck the Kogan Smart Blinds were working in Home Assistant 2021.12 however they have now stopped working in 2022.2.

I upgraded to the 2022.2 releases 3 however the still don't work.
[config_entry-tuya-a7cb720f3272791232d3c9abd038aa38.json.txt](https://github.com/tuya/tuya-home-assistant/files/8015088/config_entry-tuya-a7cb720f3272791232d3c9abd038aa38.json.txt)

**Comment by frenck** [comment]
@rackxp From your diagnostic download:

![image](https://user-images.githubusercontent.com/195327/152794615-54d610e4-40f6-4870-be4e-6402a78a8774.png)

You have the old, custom v2 integration installed. Not sure if this can cause the issue you are experiencing, however, I do know that will cause other issues.

**Comment by rackxp** [comment]
<img width="1697" alt="image" src="https://user-images.githubusercontent.com/72738248/152864516-dadaaa63-1857-4069-aae1-5fdfadbb79b7.png">
After removing the old v2 integration the issue is still present. Log file attached.
[config_entry-tuya-a7cb720f3272791232d3c9abd038aa38.json-2.txt](https://github.com/tuya/tuya-home-assistant/files/8018411/config_entry-tuya-a7cb720f3272791232d3c9abd038aa38.json-2.txt)

**Comment by rackxp** [comment]
<img width="345" alt="image" src="https://user-images.githubusercontent.com/72738248/152864948-41a7689b-8b19-4321-abd8-0321b4464e67.png">

**Comment by shortaj66** [comment]
I also have the same issu
[tuya-c1e5b3d800dc84d4338352e7c1ba18da-Cat Window-f4e213206a68736709002114d96c463f.json.txt](https://github.com/tuya/tuya-home-assistant/files/8019768/tuya-c1e5b3d800dc84d4338352e7c1ba18da-Cat.Window-f4e213206a68736709002114d96c463f.json.txt)
e

**Comment by shortaj66** [comment]
I also have the same issue
[tuya-c1e5b3d800dc84d4338352e7c1ba18da-Cat Window-f4e213206a68736709002114d96c463f.json.txt](https://github.com/tuya/tuya-home-assistant/files/8019774/tuya-c1e5b3d800dc84d4338352e7c1ba18da-Cat.Window-f4e213206a68736709002114d96c463f.json.txt)

**Comment by ionultd** [comment]
I have the same issue  as the others: tuya-iot-py-sdk == 0.6.6

**Comment by markwiese** [comment]
I have this issue. Posting my diagnostics from issue #65472. https://github.com/home-assistant/core/files/7990954/tuya-5834841df6c900b2258cba1594570946-Laundry.Blind-cf7f19ad3520ea02bed00689f77cc73a.json.txt

**Comment by rackxp** [comment]
<img width="800" alt="image" src="https://user-images.githubusercontent.com/72738248/153777512-a7c2234f-0c3a-4128-8b95-7387a35a5f17.png">
`
[config_entry-tuya-a7cb720f3272791232d3c9abd038aa38.json-3.txt](https://github.com/tuya/tuya-home-assistant/files/8056885/config_entry-tuya-a7cb720f3272791232d3c9abd038aa38.json-3.txt)
`

Can confirm that the issues Release 2022.2.6 - February 11

**Comment by comangus13** [comment]
Thanks @frenck for the response, I have updated to HA core-2022.2.6 and I can confirm tha the issue perssit. See dignistics file for both devices. Let  me know if further info required

[tuya-28788e8e2e9279549abce0a0c3abc1dd-Master Blinds-57a0b253953a64044d72e4ca39686645.json.txt](https://github.com/tuya/tuya-home-assistant/files/8067310/tuya-28788e8e2e9279549abce0a0c3abc1dd-Master.Blinds-57a0b253953a64044d72e4ca39686645.json.txt)
[tuya-28788e8e2e9279549abce0a0c3abc1dd-Brilliant Smart Sensor Light-26e3cc4601809661157a03d7e57c7113.json.txt](https://github.com/tuya/tuya-home-assistant/files/8067315/tuya-28788e8e2e9279549abce0a0c3abc1dd-Brilliant.Smart.Sensor.Light-26e3cc4601809661157a03d7e57c7113.json.txt)

**Comment by frenck** [comment]
@comangus13 For you smart light sensor, this is received from Tuya's API:


```json
    "function": {},
    "status_range": {},
    "status": {},
```

☝️ Those are the available functions and statuses Home Assistant can work with. As you can see, there is no information provided by Tuya, so Home Assistant can't do anything.

For that device, please contact Tuya directly by submitting a support ticket from your Tuya  IoT account.

The blinds however, those diagnostics have been helpful. I've located a bug and will open a fix for that one.

**Comment by ionultd** [comment]
@frenck While I notice that one of the issues you have addressed what about the Blinds issue which appears to be most of this thread? Has Tuya disabled this too?

**Comment by frenck** [comment]
@ionultd Because the main issue in this report, as reported by the author of this issue, has been solved.

All other issues stacked/added on the side in this issue, is not a concern.  If you have a different issue than reported in the openings post, please, just open a new issue instead.

Stacking issues don't work (otherwise we could have opened a single issue called "BUG"  and put just everything in that single issue instead 😉 )

../Frenck

**Comment by markwiese** [comment]
> @frenck While I notice that one of the issues you have addressed what about the Blinds issue which appears to be most of this thread? Has Tuya disabled this too?

@ionultd If you look at the PR, it appears the issue with Kogan blinds has been addressed.

**Comment by markwiese** [comment]
I installed the 2012.2.7 update, yesterday, and can confirm the Kogan blinds issue has been resolved. Thanks, @frenck :)

**Comment by comangus13** [comment]
@frenck I can confirm the kogan blinds issue is resolved in 2022.2.7, thanks for the assistance

I will create a ticket with tuya for the Sensor light issue
